src/System.Globalization/tests/CharUnicodeInfo/CharUnicodeInfoTestData.cs

```diff
@@ -0,0 +1,228 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace System.Globalization.Tests
+{
+    public static class CharUnicodeInfoTestData
+    {
+        private static List<CharUnicodeInfoTestCase> s_testCases = new List<CharUnicodeInfoTestCase>();
+
+        public static IEnumerable<object[]> TestData()
+        {
+            if (s_testCases.Count == 0)
+            {
+                string fileName = "Unicode_8_0_0_CategoryData.txt";
+                string name = typeof(CharUnicodeInfoGetUnicodeCategoryTests).GetTypeInfo().Assembly.GetManifestResourceNames().First(n => n.EndsWith(fileName, StringComparison.Ordinal));
+                Stream stream = typeof(CharUnicodeInfoGetUnicodeCategoryTests).GetTypeInfo().Assembly.GetManifestResourceStream(name);
+
+                using (var reader = new StreamReader(stream))
+                {
+                    int lineCount = 0;
+                    while (!reader.EndOfStream)
+                    {
+                        lineCount++;
+                        string line = reader.ReadLine();
+                        s_testCases.Add(Parse(line));
+                    }
+                }
+            }
+
+            foreach (CharUnicodeInfoTestCase testCase in s_testCases)
+            {
+                yield return new object[] { testCase };
+            }
+        }
+
+        public static CharUnicodeInfoTestCase Parse(string line)
+        {
+            // Data is in the format: 
+            // code-value;
+            // character-name; (ignored)
+            // general-category;
+            // canonical-combining-classes; (ignored)
+            // bidirecional-category; (ignored)
+            // character-decomposition-mapping; (ignored)
+            // decimal-digit-value; (ignored)
+            // digit-value; (ignoed)
+            // number-value;
+            string[] data = line.Split(';');
+            string charValueString = data[0];
+            string charCategoryString = data[2];
+            string numericValueString = data[8];
+
+            char codeValue = (char)int.Parse(charValueString, NumberStyles.AllowHexSpecifier);
```

Comment:
note that Unicode .txt file include a range entries
```
3400;<CJK Ideograph Extension A, First>;Lo;0;L;;;;;N;;;;;
4DB5;<CJK Ideograph Extension A, Last>;Lo;0;L;;;;;N;;;;;
```
your parsing code will still work but it will test only 2 entries in that range. I think this is fine but I just wanted to get your attention to that